engines/advancedDetector.cpp

```diff
@@ -626,6 +651,54 @@ static bool getFilePropertiesIntern(uint md5Bytes, const AdvancedMetaEngine::Fil
 	return true;
 }
 
+// Add backslash before double quotes (") and backslashes themselves (\)
+Common::String escapeString(const char *string) {
+	if (string == nullptr || Common::String(string) == "")
+		return "NULL";
+
+	Common::String res = "";
+
+	for (int i = 0; string[i] != '\0'; i++) {
+		if (string[i] == '"' || string[i] == '\\')
+			res += "\\";
+
+		res += string[i];
+	}
+
+	return res;
+}
+
+void AdvancedMetaEngineDetection::dumpDetectionEntries() const {
+	const byte *descPtr;
+
+	for (descPtr = _gameDescriptors; ((const ADGameDescription *)descPtr)->gameId != nullptr; descPtr += _descItemSize) {
+		auto g = ((const ADGameDescription *)descPtr);
+		const char *title = ((const PlainGameDescriptor *)_gameIds)->description;
+
+		debug("game (");
+		debug("\tname %s", escapeString(g->gameId).c_str());
+		debug("\ttitle %s", escapeString(title).c_str());
+		debug("\textra %s", escapeString(g->extra).c_str());
+		debug("\tlanguage %s", escapeString(getLanguageLocale(g->language)).c_str());
+		debug("\tplatform %s", escapeString(getPlatformCode(g->platform)).c_str());
+		debug("\tsourcefile %s", escapeString(getName()).c_str());
+
+		for (auto fileDesc = g->filesDescriptions; fileDesc->fileName; fileDesc++) {
+			Common::String fname = fileDesc->fileName;
+			int64 fsize = fileDesc->fileSize;
+			Common::String md5 = fileDesc->md5;
+			MD5Properties md5prop = gameFileToMD5Props(fileDesc, g->flags);
+			auto md5Prefix = Common::String(md5PropToGameFile(md5prop));
```

Comment:
Better write `Common::String md5Prefix = md5PropToGameFile(md5prop)`.

In general, this short piece of code is very messy: inconsistent usage of types, double conversion, etc.